lib/refresh.coffee

```diff
@@ -40,58 +40,105 @@ aliasItem = ($page, $item, oldItem) ->
-    {type: 'move', order: order}
-  else if moveFromPage
-    console.log 'drag from', $sourcePage.find('h1').text()
-    {type: 'remove'}
-  else if moveToPage
-    $item.data 'pageElement', $thisPage
-    $before = $item.prev('.item')
-    before = getItem($before)
-    item = aliasItem $thisPage, $item, item
-    {type: 'add', item, after: before?.id}
-  action.id = item.id
-  pageHandler.put $thisPage, action
+  destinationIsGhost = $destinationPage.hasClass('ghost')
+
+  moveWithinPage = equals($sourcePage, $destinationPage)
+  moveBetweenDuplicatePages = not moveWithinPage and \
+    $sourcePage.attr('id') == $destinationPage.attr('id')
+
+  if destinationIsGhost or moveBetweenDuplicatePages
+    $(evt.target).sortable('cancel')
+    return
+
+  if moveWithinPage
+    order = getStoryItemOrder($(this))
+    if not _.isEqual(order, $(this).data('order'))
+      pageHandler.put $destinationPage, {id: item.id, type: 'move', order: order}
+    return
+  copying = sourceIsGhost or evt.shiftKey
+  if copying
+    # If making a copy, update the temp clone so it becomes a true copy.
+    $('.shadow-copy').removeClass('shadow-copy')
+      .data($item.data()).attr({'data-id': $item.attr('data-id')})
+  else
+    pageHandler.put $sourcePage, {id: item.id, type: 'remove'}
+  # Either way, record the add to the new page
+  $item.data 'pageElement', $destinationPage
+  $before = $item.prev('.item')
+  before = getItem($before)
+  item = aliasItem $destinationPage, $item, item
+  pageHandler.put $destinationPage,
+                  {id: item.id, type: 'add', item, after: before?.id}
+
+changeMouseCursor = (e, ui) ->
+  $sourcePage = ui.item.data('pageElement')
+  sourceIsGhost = $sourcePage.hasClass('ghost')
+  $destinationPage = ui.placeholder.parents('.page:first')
+  destinationIsGhost = $destinationPage.hasClass('ghost')
+  moveWithinPage = equals($sourcePage, $destinationPage)
+  moveBetweenDuplicatePages = not moveWithinPage and \
+    $sourcePage.attr('id') == $destinationPage.attr('id')
+  copying = sourceIsGhost or (e.shiftKey and not moveWithinPage)
+  if destinationIsGhost or moveBetweenDuplicatePages
+    $('body').css('cursor', 'no-drop')
+    $('.shadow-copy').hide()
+  else if copying
+    $('body').css('cursor', 'copy')
+    $('.shadow-copy').show()
+  else
+    $('body').css('cursor', 'move')
+    $('.shadow-copy').hide()
 
 initDragging = ($page) ->
+  origCursor = $('body').css('cursor')
   options =
     connectWith: '.page .story'
     placeholder: 'item-placeholder'
     forcePlaceholderSize: true
+    delay: 150
   $story = $page.find('.story')
-  $story.sortable(options).on('sortupdate', handleDragging)
+  $story.sortable(options)
+    .on 'sortstart', (e, ui) ->
+      $item = ui.item
+      $story = $item.parents '.story:first'
+      $story.data 'order', getStoryItemOrder($story)
+      # Create a copy that we control since sortable removes theirs too early.
```

Comment:
This adds come complexity it would be cool to avoid if we can. What do we need after the sortable copy is removed?

Reply:
Once we have an always up to date model for items in the page, we won't need to manually save off the order of items on the page here. We would instead get the order from the page model during the drop operation and compare it to the one we are considering.

I'm working on that next, but wanted to get this PR wrapped up to reduce its scope. This is a short term fix to prevent unnecessary moves from being constantly recorded in the journal whenever an item is picked up, but its final location doesn't change.

Reply:
Sorry, I think I may still be a little confused here. I meant to ask why we need to do our own `.clone()` below instead of using the sortable element that we were before?

Reply:
We are still using the original sortable element. That's what gets dragged around. What we are doing here is creating a copy of the dragged element to be left behind to represent the original element. This starts out hidden and is shown when we switch from doing a 'move' to doing a 'copy. This gives the user an additional visual indication that a copy is being performed (other than just the change in the mouse cursor). On drop we either remove this hidden element (for a move) or convert it to a proper item (for a copy).

Reply:
Ahh, now I understand! Thanks for explaining.

Reply:
Ahh, now I understand! Thanks for explaining.